common/src/main/java/org/opensearch/ml/common/agent/MLToolSpec.java

```diff
@@ -119,6 +148,9 @@ public static MLToolSpec parse(XContentParser parser) throws IOException {
                 case INCLUDE_OUTPUT_IN_AGENT_RESPONSE:
                     includeOutputInAgentResponse = parser.booleanValue();
                     break;
+                case CONFIG_FIELD:
+                    configMap = getParameterMap(parser.map());
```

Comment:
may want to rename `getParameterMap` to be more general `getStringToStringMap` or similar.

Reply:
It's an existing function, I prefer to keep it unchanged at this moment.